### PR TITLE
2 add a sql database for running app locally

### PR DESCRIPTION
**Pull request by jordan-git-lewis**
The database runs locally in its current state.
Known issues:
in speak.py line 56 - DB is created here initially, but then it gets commented out to not overwrite the previous data.
in speak.py line 64 - References are written to DB here. This will execute every time the program is run, which is unnecessary

### CODE REVIEW

File: server/app/api/speak.py [hunk 1]
```diff
@@ -109,6 +111,8 @@ def write_record_route():
 
     try:
         db_helper.write_record(data, conn)
+        db_helper.write_local_record(data)
+        print("Write_Record_Route local occured")
```
Comment (aanchan) [inline server/app/api/speak.py]: I would not necessarily print things, but log using the logger module. I am not sure if this is consistent across the code base. If not, we could make a Github issue for it.

File: server/app/api/speak.py [hunk 2]
```diff
@@ -87,13 +91,11 @@ def get_records():
     Returns:
         _type_: _description_
     """
+    # conn = db_helper.get_db_connection()
```
Comment (aanchan) [inline server/app/api/speak.py]: I would remove this as it is a stray comment.

File: server/app/scripts/db_helper.py [hunk 3]
```diff
@@ -2,18 +2,45 @@
...
 # This is executed in speak.py
 
+def connect_to_local_db():
+    connection = sqlite3.connect('database.db')
+    # Changes directory to access schema table
+    # path = os.path.realpath(__file__)
+    # dir = os.path.dirname(path)
+    # dir = dir.replace('scripts', 'dbTables')
+    # os.chdir(dir)
+
```
Comment (aanchan) [inline server/app/scripts/db_helper.py]: These commented lines could be removed

File: server/app/scripts/db_helper.py [hunk 4]
```diff
@@ -2,18 +2,45 @@
...
+    with open('schema.sql') as f:
+        connection.executescript(f.read())
+
+    cursor = connection.cursor()
+    # Practice post
+    # cursor.execute("INSERT INTO audio (session_id, s3_url, date, validated, ref_id, sequence_matcher, cer, metaphone_match) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)",
+    #             ('1', '1', 'First Post', '', '1', '1', '1.0', '1.0', '1.0')
+    #             )
+    # for row in cursor.execute("select * from audio"):
+    #     print(row)
```
Comment (aanchan) [inline server/app/scripts/db_helper.py]: These commented lines could also be removed

File: server/app/scripts/db_helper.py [hunk 5]
```diff
@@ -2,18 +2,45 @@
...
+    connection.commit()
+    connection.close()
+
+    return cursor
+
 
 def connect_to_ec2():
```
Comment (aanchan) [inline server/app/scripts/db_helper.py:22]: This method does have a conflict from a base branch, and the original MYSQL connections seem to have been removed. We could have the best of both worlds here and have it connect to a local db on a later ticket.

Currently, connect_to_ec2, isn't this duplicated code which is the same as write_references_to_db?  If it is necessary, we could rename it to something more meaningful.

File: server/app/scripts/referencesDB.py
Comment (aanchan) [inline server/app/scripts/referencesDB.py:1]: Do we need this referencesDB.py gives that we read from a text file and write to the database? If it is duplicate code maybe we can delete it?

File: server/app/database.db
Comment (aanchan) [inline server/app/database.db:1]: We should probably not commit the .db file

**Review by aanchan — CHANGES_REQUESTED** [review]

**Comment by Rao2321** [comment]
1. moving all the names like database.db, schema.sql and other configurables to config file
2. validating the input before inserting into the database
3. having a check like if else if it is dev/local env and if the db exists and having else with creating db, because if X tries to clone the code and try to run it hsould ideally create the db without uncommenting
4. and some unit test cases, from my understanding there are no unittesting code for db utils or connections i see only test cases for api, data utils and validations. this also includes testing connections to aws

**Comment by jordan-git-lewis** [comment]
> 1. moving all the names like database.db, schema.sql and other configurables to config file
> 2. validating the input before inserting into the database
> 3. having a check like if else if it is dev/local env and if the db exists and having else with creating db, because if X tries to clone the code and try to run it hsould ideally create the db without uncommenting
> 4. and some unit test cases, from my understanding there are no unittesting code for db utils or connections i see only test cases for api, data utils and validations. this also includes testing connections to aws

Updated the config.json to have variable names separated by APP and DATABASE. The program now pulls from that file accordingly, but let me know if there is a better way to write this

**Comment by aanchan** [comment]
Can add a TYPE variable to config.json in the DATABASE field to:
   -  unify the connect_to_ec2 and connect_to_local_db_methods
    - And unify the write_to_db and write_local_record method

File: server/app/database.db
Comment (aanchan) [inline server/app/database.db:1]: Remove the .db file

File: server/app/scripts/db_helper.py [hunk 6]
```diff
@@ -2,15 +2,22 @@
...
 from dotenv import load_dotenv
 import os
 import boto3
 import threading
+import sqlite3
+import pathlib
+
+current_dir = pathlib.Path(__file__).parent.resolve()
+config_path = current_dir / ".." / "config.json"
+config = json.loads(config_path.read_text())
```
Comment (aanchan) [inline server/app/scripts/db_helper.py]: Add a with open here

**Pull request merged**